src/codegate/extract_snippets/body_extractor.py

```diff
@@ -0,0 +1,104 @@
+from abc import ABC, abstractmethod
+from typing import List
+
+from codegate.extract_snippets.message_extractor import (
+    AiderCodeSnippetExtractor,
+    ClineCodeSnippetExtractor,
+    CodeSnippetExtractor,
+    DefaultCodeSnippetExtractor,
+    OpenInterpreterCodeSnippetExtractor,
+)
+
+
+class BodyCodeSnippetExtractor(ABC):
+
+    def __init__(self):
+        # Initialize the extractor in parent class. The child classes will set the extractor.
+        self._snippet_extractor: CodeSnippetExtractor = None
+
+    def _extract_from_user_messages(self, data: dict) -> set[str]:
+        copied_data = data.copy()
```

Comment:
Do we need the copy here? Please double check this with me (I am really not sure) but the code snippets are retrieved using `match.group()` eventually.

The reason I am nitpicking about this is - when we ditch the `ChatCompletionRequest` through the work that @blkt is doing, we'll have methods to retrieve the messages from the request instead of a dict.

Reply:
sorry that was unclear, what I mean to say above was - I think match.group already returns a copy